### ISSUE TITLE
Implement this use case for upgrade testing

### ISSUE THREAD

**Issue by sghai** [SatelliteQE/automation-tools#532]
```
1. Create repository RepoFoo that you will later add to your Satellite. This repository should contain PackageFoo-1.0.rpm
2. Install satellite 6.1
3. Create custom product ProductFoo pointing to repository RepoFoo
4. Sync RepoFoo
5. Create content view CVFoo
6. Add RepoFoo to CVFoo
7. Publish version 1 of CVFoo
8. Upgrade satellite to 6.2.7
9. Remove PackageFoo-1.0.rpm from RepoFoo
10. Add PackageFoo-2.0.rpm to RepoFoo
11. Sync RepoFoo
12. Publish version 2 of CVFoo
13. Delete version 1 of CVFoo
14. run /etc/cron.weekly/katello-remove-orphans
15. Subscribe ClientA to CVFoo
16. Try to install PackageFoo-1.0.rpm on ClientA
17. Notice that yum thinks it's there based on the repo metadata but then fails to download it with 404
18. Try to install PackageFoo-2.0.rpm
19. Notice that yum doesn't think it's present in the repository
```

**Comment by sghai** [comment]
@sanket: as we discussed on IRC.. Please take this up on priority..

**Comment by san7ket** [comment]
@sghai , yes. I was thinking rather than doing this using hammer from scratch, we could use @rochacbruno's https://github.com/SatelliteQE/satellite-populate tool to populate entities for the test cases and scenarios. Because, as more scenarios would show up in future we would need to add more and more hammer support to automation tools and satellite-populate could be effective.

**Comment by sghai** [comment]
@san7ket : Yeah.. agreed. we don't want to do anything from scratch.. However, as we discussed about couple of other approaches I think we already have a simple solution. Lets try to implement that first and see how that goes.

**Comment by san7ket** [comment]
4 test_cases for creating a product ,2 preupgrade tests for creating product and then 2 post upgrade for validating it, I guess the basic structure for this is ready , now can start with implementing the use-case.

>  pytest test_scenerio.py -s -m pre_upgrade -v 
> test session starts 
> platform linux2 -- Python 2.7.12, pytest-3.0.6, py-1.4.31, pluggy-0.4.0 -- /usr/bin/python
> cachedir: ../.cache
> 
> test_scenerio.py::TestUserCase::test_create_product {'TestUserCase': {'id': 90, 'name': u'TZryWorvkB'}}
> PASSED
> test_scenerio.py::Case::test_create_product1 {'Case': {'id': 91, 'name': u'ATZXhNqRKz'}, 'TestUserCase': {'id': 90, 'name': u'TZryWorvkB'}}
> PASSED
> == 2 tests deselected 2 passed, 2 deselected in 26.63 seconds 
> pytest test_scenerio.py -s -m post_upgrade
> ===================================================================================== test session starts 
> platform linux2 -- Python 2.7.12, pytest-3.0.6, py-1.4.31, pluggy-0.4.0
> rootdir: /home/sjagtap/PycharmProjects/automation-tools, inifile:
> plugins: cov-2.3.1, xdist-1.15.0
> collected 4 items
> 
> test_scenerio.py ..
> == 2 tests deselected  2 passed, 2 deselected in 11.40 seconds 

===========================================================================

**Comment by san7ket** [comment]
Cannot pull custom repo's in docker clients, The issue is weird , as when the docker client is registered with satellite's ak, and when we do a repolist , the clients doesn't fetch the repos immediately , after waiting for half hour or so , we are able to fetch the custom repos back, 
What all I tried,
1. There are no redhat repos and no valid subs 
2. There are redhat repos and valid with custom repos , Only the redhat repos are pulled 
3. Checked the firewall , Done subscription refresh 
4. No errors are logged , so no help there 
5. Subscription manager work's fine, when subscribing with no subs , it shows invalid status , when subscribing with valid subs , it shows valid subscription

I am using https://github.com/JacobCallahan/content-host-d , to create and subscribe docker clients to satellite
